reconcile/utils/terrascript_aws_client.py

```diff
@@ -288,13 +288,41 @@ def __init__(
 
             ts += provider.template(version="2.2.0")
 
+            access_key_backend_value = config["aws_access_key_id"]
+            secret_key_backend_value = config["aws_secret_access_key"]
+            bucket_backend_value = ""
+            key_backend_value = ""
+            region_backend_value = ""
+
+            for filtered_account in filtered_accounts:
+                if filtered_account.get("name") == name:
+                    terraform_state = filtered_account.get("terraformState")
+                    if not terraform_state:
+                        bucket_backend_value = config["bucket"]
+                        key_backend_value = config["{}_key".format(integration)]
+                        region_backend_value = config["region"]
+                    else:
+                        tf_state_integration = terraform_state["integrations"]
+                        for key in tf_state_integration:
+                            key_value = key.get("key")
+                            integration_value = key.get("integration")
+                            transformed_integration = str(integration_value).replace(
+                                "-", "_"
+                            )
+                            if transformed_integration == self.integration:
+                                bucket_backend_value = terraform_state.get("bucket")
+                                key_backend_value = str(key_value)
+                                region_backend_value = terraform_state.get("region")
+                            else:
+                                continue
+
```

Comment:
consider using the `bucket`, `region` and `key` fields from the account if no 'key' is defined within `terraformState`.
or fail the integration if you `terraformState` is present but no `key` is defined for an integration.

Reply:
with "no key defined" i meant if the terraformState does not contain information for an integration, try to discover the required information the old way